src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,511 @@
+                     logPrefix);
+        return false;
+    }
+
+    if (!ptr->getParameter("gait_start_with_left", pImpl->gait.startWithLeft))
+    {
+        log()->info("{} Using default gait_start_with_left={}.",
+                    logPrefix,
+                    pImpl->gait.startWithLeft);
+    }
+
+    if (!ptr->getParameter("gait_terminal_step", pImpl->gait.terminalStep))
+    {
+        log()->info("{} Using default gait_terminal_step={}.", logPrefix, pImpl->gait.terminalStep);
+    }
+
+    if (!ptr->getParameter("gait_reset_starting_foot_if_still",
+                           pImpl->gait.resetStartingFootIfStill))
+    {
+        log()->info("{} Using default gait_reset_starting_foot_if_still={}.",
+                    logPrefix,
+                    pImpl->gait.resetStartingFootIfStill);
+    }
+
+    pImpl->planner->startWithLeft(pImpl->gait.startWithLeft);
+    pImpl->planner->addTerminalStep(pImpl->gait.terminalStep);
+    pImpl->planner->resetStartingFootIfStill(pImpl->gait.resetStartingFootIfStill);
+
+    // =============================
+    // UnicyclePlanner configuration
+    // =============================
+
+    if (!okPlanner)
+    {
+        log()->error("{} Failed to configure UnicyclePlanner.", logPrefix);
+        return false;
+    }
+
+    return true;
+}
+
+const Planners::UnicyclePlannerState& Planners::UnicyclePlanner::getOutput() const
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::getOutput]";
+
+    pImpl->state = UnicyclePlannerState();
+
+    if (!this->isOutputValid())
+    {
+        return pImpl->state;
+    }
+
+    auto convertStep
+        = [&](const ::Step& step, const ::Step& nextStep) -> const Contacts::PlannedContact {
+        Contacts::PlannedContact contact;
+        contact.name = step.footName;
+        contact.activationTime = step.impactTime;
+
+        contact.pose.quat(Eigen::AngleAxisd(step.angle, Eigen::Vector3d::UnitZ()));
+        contact.pose.translation({step.position[0], step.position[1], 0.0});
+
+        // Note: we use the impact time to compute the duration of the step.
+        //       If the step is the last one, a default 5 seconds duration is used.
+        const double duration = nextStep.impactTime - step.impactTime;
+        assert(duration >= 0);
+        contact.deactivationTime
+            = duration > 0.0 ? contact.activationTime + duration * pImpl->gait.stancePhaseRatio
+                             : contact.activationTime + 5.0;
+
+        return contact;
+    };
+
+    auto convertSteps
+        = [&convertStep](decltype(
+              ::FootPrint().getSteps())& steps) -> std::optional<const Contacts::ContactList> {
+        Contacts::ContactList list;
+
+        for (std::size_t i = 0; i < steps.size(); ++i)
+        {
+            // Last step
+            if (i == steps.size())
+            {
+                if (!list.addContact(convertStep(steps[i], steps[i])))
```

Comment:
Isn't this failing because of the assert on the duration?

Reply:
The assert is `>= 0`, it shouldn't fail but I'm not sure if this is what you meant.

Reply:
Sorry, you are right, I misread it!